internal/vault/vault.go

```diff
@@ -464,3 +513,16 @@ func (v *Vault) IsVaultInitializedAndUnsealed() error {
 
 	return nil
 }
+
+func (v *Vault) addVaultNamespaceToRequest(request *vault.Request) {
+	vaultIssuer := v.issuer.GetSpec().Vault
+	if vaultIssuer != nil && vaultIssuer.Namespace != "" {
+		if request.Headers != nil {
+			request.Headers.Add("X-VAULT-NAMESPACE", vaultIssuer.Namespace)
+		} else {
+			vaultReqHeaders := http.Header{}
+			vaultReqHeaders.Add("X-VAULT-NAMESPACE", vaultIssuer.Namespace)
+			request.Headers = vaultReqHeaders
+		}
+	}
+}
```

Comment:
@maelvls can you remove this unused function?

Reply:
Done in 5083b3e36